### PR TITLE
Add Spellcheck to EditableText (Android)

### PR DESCRIPTION
**Pull request by camsim99**
Same as https://github.com/flutter/flutter/pull/104460.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/spell_check.dart [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+      text: text.substring(start, composingRegion.start)
+    )
+  );
+  treeChildren.add(
+    TextSpan(
+      style: composingTextStyle,
+      text: text.substring(composingRegion.start, composingRegion.end)
+    )
+  );
+}
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/spell_check.dart:330]: I don't see a problem with keeping everything here in its own file in the widgets library :+1: .  Unless @camsim99 or another reviewer has a better idea.

**Review by justinmc — APPROVED** [review]
LGTM with nits.  Mainly just the question about making some functions private.

File: packages/flutter/lib/src/widgets/spell_check.dart (hunk 1 above)
Comment (camsim99) [reply packages/flutter/lib/src/widgets/spell_check.dart:330]: Will definitely be useful to group the methods together especially when the menu logic needs to be added somewhere.

File: packages/flutter/lib/src/widgets/spell_check.dart [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+// Methods for displaying spell check results:
+
+/// Adjusts spell check results to correspond to [newText] if the only results
+/// that the handler has access to are the [results] corresponding to
+/// [resultsText].
+///
+/// Used in the case where the request for the spell check results of the
+/// [newText] is lagging in order to avoid display of incorrect results.
+List<SuggestionSpan> correctSpellCheckResults(
```
[the remark this answers is not in the document]
Comment (justinmc) [reply packages/flutter/lib/src/widgets/spell_check.dart]: No worries.  It looks good now that you made both functions private.  Now there is only one public function (buildTextSpanWithSpellCheckSuggestions) which is pretty benign.

**Comment by camsim99** [comment]
@cyanglaz Would you be willing to give this a second look?

**Review by cyanglaz — APPROVED** [review]
LGTM

**Review by 007hz — COMMENTED** [review]
>

**Pull request merged**